x-pack/platform/plugins/shared/cases/server/client/alerts/get.ts

```diff
@@ -37,3 +42,41 @@ export const getAlerts = async (
     ...alert._source,
   }));
 };
+
+export const getAlertMetadataFromComments = async (
+  comments: AttachmentAttributes[],
+  clientArgs: CasesClientArgs
+): Promise<CaseMetadata> => {
+  const alertInfo: AlertInfo[] = comments.flatMap((c) => {
+    if (c.type === AttachmentType.alert) {
+      const { ids, indices } = getIDsAndIndicesAsArrays(c);
+      return ids.map((alertId, index) => ({ id: alertId, index: indices[index] }));
+    }
+    return [];
+  });
+
+  const alertsDocs = await getAlerts(alertInfo, clientArgs);
+
+  const metadata = alertsDocs.reduce<Record<string, Set<string>>>(
+    (acc, alert) => {
+      const { [TAGS]: tags = [], [ALERT_GROUPING]: grouping } = alert;
+
+      tags.forEach((t) => {
+        acc.tags.add(t);
+      });
+
+      if (grouping) {
+        Object.entries(flattenObject(grouping)).forEach(([k, v]) => {
+          if (typeof v !== 'string') return;
+          (acc[k] ??= new Set()).add(v);
+        });
+      }
+      return acc;
+    },
+    { tags: new Set() }
```

Comment:
This metadata currently only returns tags and grouping? Is the idea to make this a shared utility in the future? Just thinking out loud, but should solutions be able to define what the metadata fields they want are or is the plan to have a single metadata architecture that all solutions will end up using?